### PR TITLE
Group modal sidebar label field aggregations

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

When latency is a concern for the App server, grouping the modal sidebar requests (aggregations for label fields) into two distinct network calls, samples and frames (if applicable), improves the loading experience.

This behavior is only targeted for the modal, where aggregation requests are only with respect to a sample or group.

## How is this patch tested? If it is not, please explain why.

Recoil unit tests

## Release Notes

* Optimized the expanded sample view sidebar

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced aggregation handling in the app to support different paths based on modal conditions.
	- Introduced functionality to access label paths based on their expansion status.
- **Refactor**
	- Improved the logic for filtering and finding within the aggregation selector.
	- Renamed and modified selectors for better clarity and functionality.
- **Tests**
	- Added test cases for aggregation path accumulation functionality.

### CODE REVIEW

**Comment by swheaton** [comment]
experimental ai reviewer ...

@coderabbitai review

**Comment by swheaton** [comment]
hit rate limit last time
@coderabbitai review

File: app/packages/state/src/recoil/aggregations.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { describe, expect, it, vi } from "vitest";
+vi.mock("recoil");
+vi.mock("recoil-relay");
+
+import { setMockAtoms, TestSelectorFamily } from "../../../../__mocks__/recoil";
+import * as aggregations from "./aggregations";
+import { State } from "./types";
+
+describe("test aggregation path accumulation", () => {
```
Comment (imanjra) [inline app/packages/state/src/recoil/aggregations.test.ts:9]: Nice!

**Review by imanjra — COMMENTED** [review]
Tested locally and mostly things are looking good. Nice work.

Issue I found while testing locally:
- Expanding frames.id seems to crash the app

image

**Comment by benjaminpkane** [comment]
> Tested locally and mostly things are looking good. Nice work.
> 
> Issue I found while testing locally:
> 
> * Expanding frames.id seems to crash the app

Thank you! I will debug and address

**Review by imanjra — APPROVED** [review]
Testes again locally. The issue is fixed now. Thanks. LGTM

**Pull request merged**